.github/CONTRIBUTING.md

```diff
@@ -2,60 +2,110 @@
+
+Now, while you're working on the code, the editor will support you. Also, you can use "Code > Reformat Code" (<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>) from time to time.
+
+## Checkstyle
+As an additional measure, we use [Checkstyle](http://checkstyle.sourceforge.net/).
+You can run the checks defined in [config/checkstyle/checkstyle.xml](https://github.com/mapzen/LOST/blob/master/config/checkstyle/checkstyle.xml) either manually on the command line via `./gradlew checkstyle` or pick the task from the Gradle pane before pushing.
+It will generate a HTML file at `lost/build/reports/checkstyle/checkstyle.html` inside the project folder, listing all violations.
+You could also add a git pre commit hook.
+
+A maybe even more comfortable way is to install the [CheckStyle-IDEA plugin](https://plugins.jetbrains.com/plugin/1065) via "File->Settings->Plugins->Install JetBrains plugin".
+After the restart, navigate to "File > Settings > Other Settings > Checkstyle", click on the green <kbd>+</kbd> button on the right of the upper list, choose the checkstyle.xml from the project directory, select "Store relative to project location" and give it a description like "Mapzen LOST".
+Above the list, you can check all three additional checkboxes.
+
+Now, while editing code, failed checkstyle rules will show up like the usual Android Linter remarks.
+Also, there is a new tab/pane "Checkstyle" in the lower edge of the GUI, allowing to manually run the checks on the current or all files in the project, you can jump to the location by double clicking the entry.
+Additionally, you can select "Scan with Checkstyle" in the "Commit" dialogue.
+
+Checkstyle remarks should be fixed before issuing a pull request.
+
+## Managing upstream changes
+While you work on your feature branch, there may be changes happening on the upstream master. It is helpful to monitor them, by fetching from time to time.
+Please do not merge them into your branch, but instead [rebase your changes](https://git-scm.com/book/en/v2/Git-Branching-Rebasing) before issuing the pull request.
+Personally we prefer rebase to merge commits since it makes changes easier to review and history a bit more tidy. 
+You can do the rebase from the command line or, in Android Studio, while you are on your branch, select "Rebase onto" from the context menu of the upstream master in the "Log" of the "Version Control" tab.
+You will probably (depending on how big the master changes and your's are) have to "Resolve conflicts" in the files separately (select from the context menu from the individual files shown in red under "Log", always finish by "Merge"), and afterwards select "Skip Commit in Rebasing" from the "VCS->Git" menu.
+More information on this is available at [IntelliJ](https://www.jetbrains.com/help/idea/2016.2/interactive-rebase.html) or [Youtube](https://www.youtube.com/watch?v=lILdMGa-r38).
```

Comment:
I think we could do without this video link. The other resources provided here seem sufficient. And while the video is somewhat helpful it doesn't really cover rebasing until the very end (after showing how to create merge commits - which is what we want to avoid).